ambry-router/src/main/java/com.github.ambry.router/GetBlobInfoOperation.java

```diff
@@ -46,7 +48,17 @@
  * which is either the only chunk in the case of a simple blob, or the metadata chunk in the case of composite blobs.
  */
 class GetBlobInfoOperation extends GetOperation {
```

Comment:
I'm having some trouble understanding the interactions in this class and the way they are modelled. This is the outline that makes sense to me

1) Send request(s)
2) Handle response. If error is during deserialization do nothing since `poll()` will called right away.
3) If no error in 2 and no decryption required, set result and call `checkAndMaybeComplete()`
(until here is the code that already exists)
4) If no error in 2 and decryption is required, set off a decryption job (make sure the next `poll()` does _not_ send any new requests)
5) If decryption job succeeds, set result and call `checkAndMaybeComplete()` in the callback itself
6) If decryption fails, reset state and call `onPollReady()` so that so that `poll()` is called.

Reply:
yeah. Lets see if this makes sense.

```

1.    Send request(s)
2.    Handle response. If error is during deserialization, operationException is set within handleResponse(). (check the line where processGetBlobInfoResponse(getRequestInfo, getResponse) is called. ) and then call checkAndMaybeComplete()
3.    If no error in 2 and no decryption required, set result and call checkAndMaybeComplete()
    (until here is the code that already exists)
4.    If no error in 2 and decryption is required, set off a decryption job (make sure the next poll() does not send any new requests)
5.    If decryption job succeeds, set result and call checkAndMaybeComplete() in the callback itself
6.    If decryption fails, set operationexception and call checkAndMaybeComplete()

```

Reply:
Wanted to remind you something. 

Once deserialization succeeds from one server response, we don't process any other responses. We go ahead with whatever happens with the successfully deserialized response. If decryption fails, the operation fails. I did think of that route before (keep accepting more responses until both deserialization and decryption succeeds), but after discussions with priyesh and casey, came up with this conclusion that once deserialization succeeds, we don't care for other responses. Also, the code might get too complicated if we take that approach.

Reply:
`Once deserialization succeeds from one server response, we don't process any other responses.` - this is true before encryption. Failure of decryption is a failure is it not? Why wouldn't we want more responses?